docs/src/content/docs/challenges/angular/4-context-outlet-typed.md

```diff
@@ -16,30 +16,34 @@ sidebar:
 
 ## Information
 
-Angular offer the static function [`ngTemplateContextGuard`](https://angular.io/guide/structural-directives#typing-the-directives-context) to strongly type structural directive.
+You can improve template type checking for custom directives by adding template guard properties to your directive definition. Angular offer the static function [`ngTemplateContextGuard`](https://angular.io/guide/structural-directives#typing-the-directives-context) to strongly type structural directives.
 
 However the context of **NgTemplateOutlet** type is **Object**. But with the help of the above guard, we can improve that behavior.
 
+If you’re not very familiar with `ngTemplateOutlet`, check out this article.
+
+- [NgTemplateOutlet Type Checking](https://medium.com/@thomas.laforge/ngtemplateoutlet-type-checking-5d2dcb07a2c6)
```

Comment:
You don't need this here, as it is already linked in the article button.

Reply:
Check the [PR description](https://github.com/tomalaforge/angular-challenges/pull/705#issue-2200612076), i found this article link after i finished the challenge. And in other docs, e. g. [Structural Directive](https://angular-challenges.vercel.app/challenges/angular/6-permissions/) there are links in the docs too.

Reply:
Yes, you aren't supposed to read the solution before you try the challenge.  Not all challenges have articles and the articles can help people when there are stuck or lost.     

It is better to add a note in another section about articles i.e. if you get stuck or can't complete the challenge, read a detailed solution by clicking the article button.

Reply:
Okay i'll remove the link, nevertheless imo your arguments doesn't make much sense to me. It's upon you if you take the challenge serious and don't look for a solution right away. The community solutions are available any way, for me a challenge itself is about know how to find those informations. And thus find a solution for a problem.

As i was doing this challenge I've also used google to find informations about this topic, and found thomas article about it. But anyway, your point is valid too, we don't need to make it too easy to find the answers.

Reply:
Well, googling and now prompting are part of the challenge and life of development.  The best challenges and topics are when there is incomplete and poor documentation out there.  This repo can fill the gap.

Reply:
well said 🚀